model/Bom.js

```diff
@@ -38,11 +41,46 @@ class Bom extends CycloneDXObject {
     if (pkg) {
       this._metadata = this.createMetadata(pkg, componentType)
       this._components = this.listComponents(pkg, lockfile)
+      this._dependencies = this.listDependencies(pkg)
     } else {
       this._components = []
+      this._dependencies = []
     }
   }
 
+  listDependencies (pkg) {
+    const list = []
+    this.createDependency(pkg, list)
+    return list
+  }
+
+  createDependency (pkg, list) {
+    // read-installed with default options marks devDependencies as extraneous
+    // if a package is marked as extraneous, do not include it as a dependency
+    if (pkg.extraneous) return
```

Comment:
:x: would a switch make sense here? 
we might want to include dev-dependencies of a root package, if the CLI switch for it was set
`-d, --include-dev          Include devDependencies (default: false)`

Reply:
What would be the other case in switch? from the CLI if -d or --include-dev is specified then extraneous is not set to true and dev dependencies will be considered in the BoM.

Reply:
that is good to know. 
could you document this in a comment?

Reply:
Sure